elasticsearch-api/utils/thor/generate_api.rb

```diff
@@ -35,22 +35,22 @@ module Utils
       #
       def __parse_java_source(path)
         path  += '/' unless path =~ /\/$/ # Add trailing slash if missing
-        prefix = "src/main/java/org/elasticsearch/rest/action"
+        prefix = "core/src/main/java/org/elasticsearch/rest/action"
 
-        java_rest_files = Dir["#{path}#{prefix}/**/*.java"]
+        java_rest_files = Dir["#{path}#{prefix}/**/Rest*Action.java"]
 
         map = {}
 
         java_rest_files.sort.each do |file|
           content = File.read(file)
           parts   = file.gsub(path+prefix, '').split('/')
-          name    = parts[0, parts.size-1].reject { |p| p =~ /^\s*$/ }.join('.')
+          name    = parts[-1].scan(/Rest(.+?)[A-Z]*Action\.java/).first.first.downcase + '_' + parts[-2]
```

Comment:
This was just to make sure all generate json filenames are unique, and nothing is being swallowed / overridden. I will have to think more about naming, but I start to have doubts I can find a consistent naming code, since actual naming in https://github.com/elastic/elasticsearch/tree/master/rest-api-spec/src/main/resources/rest-api-spec/api seems to not be consistent with respective [java files](https://github.com/elastic/elasticsearch/tree/master/core/src/main/java/org/elasticsearch/rest/action).